rs/ic-observability/multiservice-discovery/src/definition.rs

```diff
@@ -143,13 +144,27 @@ impl TestDefinition {
     }
 
     /// Syncs the registry update the in-memory cache then stops.
-    pub async fn sync_and_stop(&self) {
-        let _ = self.running_def.initial_registry_sync().await;
-        // if self.initial_registry_sync().await.is_err() {
-        // FIXME: Error has been logged, but ideally, it should be handled.
-        // E.g. telemetry should collect this.
-        // return;
-        // }
+    pub async fn sync_and_stop(&self, skip_update_local_registry: bool) {
+        if let Err(e) = if skip_update_local_registry {
+            sync_local_registry(
```

Comment:
@sasa-tomic just want to have this check:
```
    if use_current_version && latest_version != ZERO_REGISTRY_VERSION {
        debug!(log, "Skipping syncing with registry, using local version");
        return Ok(());
    } else if use_current_version {
        warn!(
            log,
            "Unable to use current version of registry since its a zero registry version"
        );
    }
```

Reply:
makes sense? :)